### ISSUE TITLE
Support --mlock on the command line. Also there are undocumented model file parameters

### ISSUE THREAD

**Issue by sdmorrey** [ollama/ollama#6200]
### What is the issue?

The original issue was that my Mac was experiencing an exponential slow down. Upon investigating the issue I noticed my RAM usage was very, very low, but my swap was climbing at the rate of about 1GB per 1K of filled context.

In the end I discovered the --mlock flag in llama.cpp. 
I was in discord asking for help setting it since the command line Ollama straight up rejects it.

Eventually we discovered that this is controlled by the model file parameter use_mlock

Except that appears to be undocumented. In the same space I found a few other undocumented parameters.

So in a nutshell, it would lovely if you could add --mlock to the command line since I don't want to have to edit every model file. Also if you could update the docs so use_mlock and the other undocumented parameters are documented, that would be awesome.

Thanks for your hard work and such a great product!

### OS

macOS

### GPU

Intel

### CPU

Intel

### Ollama version

0.3.3

**Comment by sdmorrey** [comment]
Forgot to mention, I'll be glad to submit a PR if that would be welcome. I just want to make sure it's an oversight and not a design decision for reasons I'm just not aware of. 

Thanks!
